scripts/hooks.py

```diff
@@ -0,0 +1,301 @@
+    """Generate llms.txt file following llmstxt.org standards."""
+    ordered_files = get_ordered_files(nav_config, docs_dir)
+
+    # Generate llms.txt content
+    llms_txt_content = []
+
+    # Add header following MCP format
+    llms_txt_content.append("# any-agent")
+    llms_txt_content.append("")
+    llms_txt_content.append("## Docs")
+    llms_txt_content.append("")
+
+    # Add individual file entries with better formatting
+    for file_path in ordered_files:
+        # Convert markdown path to .txt file path with full URL
+        txt_url = f"{BASE_URL}{file_path.replace(MARKDOWN_EXTENSION, TEXT_EXTENSION)}"
+
+        title = create_file_title(file_path)
+        description = get_file_description(file_path, docs_dir)
+
+        # Format entry like MCP, but link to .txt files with full URLs
+        if description:
+            llms_txt_content.append(f"- [{title}]({txt_url}): {description}")
+        else:
+            llms_txt_content.append(f"- [{title}]({txt_url})")
+
+    # Write the llms.txt file
+    llms_txt_dest = site_dir / "llms.txt"
+    write_file_content(llms_txt_dest, "\n".join(llms_txt_content))
+
+
+def generate_llms_full_txt(docs_dir, site_dir, nav_config):
+    """Generate llms-full.txt by concatenating all markdown documentation."""
+    ordered_files = get_ordered_files(nav_config, docs_dir)
+
+    # Generate the llms-full.txt content
+    llms_full_content = []
+
+    # Add header
+    llms_full_content.extend(
+        [
+            "# any-agent Documentation",
+            "",
+            "> Complete documentation for any-agent - A Python library providing a single interface to different agent frameworks.",
+            "",
+            "This file contains all documentation pages concatenated for easy consumption by AI systems.",
+            "",
+            "---",
+            "",
+        ]
+    )
+
+    # Process each markdown file
+    for file_path in ordered_files:
+        full_path = docs_dir / file_path
+
+        if full_path.exists():
+            content = read_file_content(full_path)
+            if content is not None:
+                # Clean and process content
+                cleaned_content = clean_markdown_content(content, file_path)
+
+                # Add section separator
+                llms_full_content.extend(
+                    [f"## {file_path}", "", cleaned_content, "", "---", ""]
+                )
+
+    # Write the combined content to llms-full.txt
+    llms_full_txt_dest = site_dir / "llms-full.txt"
+    write_file_content(llms_full_txt_dest, "\n".join(llms_full_content))
+
+
+def on_post_build(config, **kwargs):
+    """Generate llms.txt and llms-full.txt files, and serve markdown as text."""
+    docs_dir = Path(config["docs_dir"])
+    site_dir = Path(config["site_dir"])
+
+    # Get navigation configuration
+    nav_config = config.get("nav", [])
+
+    # Serve markdown files as text
+    serve_markdown_as_text(docs_dir, site_dir, nav_config)
```

Comment:
Being really picky, I'd rename this as `convert_markdown_to_text` since the serving is a separate process from doc generation (like if some param was being given to `mkdocs serve` to set the content type while serving).